isis/tests/SpiceRotationTests.cpp

```diff
@@ -0,0 +1,822 @@
+//
+// This is written as a fixture to ensure that the test kernels get unloaded
+// regardless of how the test finishes.
+class SpiceRotationKernels : public ::testing::Test {
+  protected:
+    vector<QString> kernels;
+    double startTime;
+    double endTime;
+    int frameCode;
+    int targetCode;
+
+  void SetUp() {
+    startTime = -69382819.0;
+    endTime = -69382512.0;
+    frameCode = -94031;
+    targetCode = 499;
+
+    QString dir = FileName("$ISISTESTDATA/isis/src/base/unitTestData/kernels").expanded() + "/";
+    kernels.clear();
+    kernels.push_back(dir + "naif0007.tls");
+    kernels.push_back(dir + "MGS_SCLKSCET.00045.tsc");
+    kernels.push_back(dir + "moc13.ti");
+    kernels.push_back(dir + "moc.bc");
+    kernels.push_back(dir + "moc.bsp");
+    kernels.push_back(dir + "de405.bsp");
+    kernels.push_back(FileName("$base/kernels/pck/pck00009.tpc").expanded());
+    kernels.push_back(dir + "mocSpiceRotationUnitTest.ti");
+    kernels.push_back(dir + "ROS_V29.TF");
+    kernels.push_back(dir + "CATT_DV_145_02_______00216.BC");
+    for (QString& kernel : kernels) {
+      furnsh_c(kernel.toLatin1().data());
+    }
+  }
+
+  void TearDown() {
+    for (QString& kernel : kernels) {
+      unload_c(kernel.toLatin1().data());
+    }
+  }
+};
+
+class SpiceRotationIsd : public ::testing::Test {
+  protected:
+    json isd;
+    json isdAv;
+    json isdConst;
+
+  void SetUp() {
+    isd = {{"ck_table_start_time"    , 0.0},
+           {"ck_table_end_time"      , 3.0},
+           {"ck_table_original_size" , 4},
+           {"ephemeris_times"      , {0.0, 1.0, 2.0, 3.0}},
+           {"time_dependent_frames" , {-94031, 10014, 1}},
+           {"quaternions"         , {{0.0, 0.0, 0.0, 1.0},
+                                     {-1.0 / sqrt(2), 0.0, 0.0, 1.0 / sqrt(2)},
+                                     {0.0, 1.0 / sqrt(2), 1.0 / sqrt(2), 0.0},
+                                     {-0.5, -0.5, 0.5, 0.5}}}};
+    isdAv = isd;
+    isdAv["angular_velocities"] = {{-Isis::PI / 2, 0.0, 0.0},
+                                   {0.0, Isis::PI, 0.0},
+                                   {0.0, 0.0, Isis::PI / 2},
+                                   {0.0, 0.0, Isis::PI / 2}};
+    isdConst = isd;
+    isdConst["time_dependent_frames"] = {-94030, 10014, 1};
+    isdConst["constant_frames"] = {-94031, -94030};
+    isdConst["constant_rotation"] = {1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0};
+  }
+};
+
+TEST_F(SpiceRotationKernels, FromSpice) {
+  SpiceRotation rot(frameCode);
+
+  // Start time
+  rot.SetEphemerisTime(startTime);
+  EXPECT_DOUBLE_EQ(rot.EphemerisTime(), startTime);
+
+  vector<double> startCJ = rot.Matrix();
+  EXPECT_PRED_FORMAT3(AssertVectorsNear, startCJ,
+                      (vector<double>{-0.87506927,  0.25477955, -0.41151081,
+                                       0.011442263, 0.86088548, 0.50867009,
+                                       0.48386242,  0.44041295, -0.75624969}),
```

Comment:
The strange `(vector<double>{ ... })` syntax is required because initializer lists don't expand properly in macros; the macro ignores the `{}` and thinks that each element in the list is a separate argument.